[ready] General documentation improvements

vishwakftw opened the pull request:
This closes the proposal in https://github.com/pytorch/pytorch/issues/5432.

The idea of the PR is to improve existing documentation.

### Changes made:
1. Modify Tensor operations docs.
2. Modify `torch.nn*` docs (excluding `functional`).

### Types of Changes made:
1. Fix math errors
2. Add formulae
3. Improve description if required

If these changes are acceptable, I could look at the rest of the documentation and clean it up in the same way.

vishwakftw committed:
Improvize documentation
1. Add formula for erf, erfinv
2. Make exp, expm1 similar to log, log1p
3. Symbol change in ge, le, ne, isnan

vishwakftw committed:
Fix minor nit in the docstring

vishwakftw committed:
More doc improvements
1. Added some formulae
2. Complete scanning till "Other Operations" in Tensor docs

goldsborough commented on torch/random.py:
```diff
@@ -15,12 +15,12 @@ def set_rng_state(new_state):
 
 def manual_seed(seed):
-    r"""Sets the seed for generating random numbers. And returns a
+    r"""Sets the seed for generating random numbers. This returns a
```
Nit: Just "Returns ..." is fine

vishwakftw replied:
Oops, I forgot to change it in the new commit. Will do it in the next batch.

vishwakftw committed:
Add more changes
1. Modify all torch.Tensor wherever required

vishwakftw committed:
Fix Conv docs
1. Fix minor nits in the references for LAPACK routines

ezyang commented:
Lint is failing

vishwakftw committed:
Improve Pooling docs
1. Fix lint error

ezyang commented:
Still failing lint.

```
./torch/nn/modules/pooling.py:116:1: W293 blank line contains whitespace
./torch/nn/modules/pooling.py:640:121: E501 line too long (123 > 120 characters)
./torch/nn/modules/pooling.py:673:1: W293 blank line contains whitespace
```

Note that you can run flake8 locally to see if you've got all the fixes you need.

vishwakftw commented:
Sorry about that. I forgot to add the lint checker in my pre-commit hook. I will fix them in the upcoming commits.

vishwakftw committed:
Improve docs for RNN, Normalization and Padding
1. Fix flake8 error for pooling

vishwakftw committed:
Final fixes for torch.nn.* docs.
1. Improve Loss Function documentation
2. Improve Vision Layers documentation

vishwakftw committed:
Merge branch 'master' into improve-docs-1

vishwakftw committed:
Fix lint error

SsnL commented:
@pytorchbot retest this please

vishwakftw committed:
Improve docstrings in torch.nn.init

vishwakftw committed:
Merge branch 'master' into improve-docs-1

vishwakftw committed:
Fix lint error

vishwakftw committed:
Fix minor error in torch.nn.init.sparse

vishwakftw commented:
@apaszke Should we add the images for activation functions generated in `docs/source/_static/img/activation`. There seems to be a problem with the docs while trying to render these images from the PR in https://github.com/pytorch/pytorch/pull/5457.

I am also fixing some math errors in `nn.modules.activation`.

vishwakftw committed:
Fix Activation and Utils Docs
1. Fix Math Errors
2. Add explicit clean to Makefile in docs to prevent running graph generation script
while cleaning
3. Fix utils docs

vishwakftw commented on docs/Makefile:
```diff
@@ -13,7 +13,7 @@ help:
 
 figures:
-	python source/scripts/build_activation_images.py
+	python3 source/scripts/build_activation_images.py
```
Is it possible to find out the version of Python installed for this?

ezyang replied:
Oh, that's awkward. So this is really Python 3 only? It is a basically safe assumption to assume python 3 is installed with a `python3` symlink, but it would be even better if this wasn't a Python 3 only thing.

vishwakftw replied:
I built `torch` with Python3. This is what I did: `sudo python3 setup.py build develop`.

Maybe because I am explicitly specifying `python3` is causing this to happen.

apaszke replied:
It's not Python 3 only, it's "the python you used to build PyTorch" only. I can see two possibilities:
- make the Python command to use a Makefile variable (that can be overriden using an env variable). We already use this strategy in `test/run_test.sh`.
- Do `python <script_path> || python3 <script_path>` to handle failures automatically. A bit non-obivous, so I'd lean towards the first solution

vishwakftw committed:
Make PYCMD a Makefile argument, clear up prints in the build_activation_images.py

vishwakftw commented on docs/Makefile:
```diff
@@ -7,13 +7,14 @@ SPHINXBUILD   = sphinx-build
 
 figures:
-	python3 source/scripts/build_activation_images.py
+	@$(PYCMD) source/scripts/build_activation_images.py
```
I have made `PYCMD` a Makefile argument.

apaszke replied:
Why the `@` at the beginning of the line?

vishwakftw replied:
I thought it is not required to print that command. If it is otherwise, I can change it.

apaszke replied:
Right, I forgot it silences `make`. It's ok to leave it

vishwakftw committed:
Merge branch 'master' into improve-docs-1

vishwakftw committed:
Fix batch norm doc error

vishwakftw committed:
Merge branch 'improve-docs-1' of https://github.com/vishwakftw/pytorch into improve-docs-1

vishwakftw committed:
Merge branch 'master' into improve-docs-1

vishwakftw commented:
Fixed https://github.com/pytorch/pytorch/issues/5625 in the latest commit. @SsnL

soumith committed:
Merge branch 'master' into improve-docs-1

vishwakftw commented:
@apaszke @soumith Could I clean up `torch.nn.functional` and `torch.optim` too?

soumith commented:
@vishwakftw yes please go for it :)
